src/axom/klee/ShapeSet.cpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2017-2021, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "axom/klee/ShapeSet.hpp"
+
+#include <utility>
+#include <stdexcept>
+
+#include "axom/core/utilities/FileUtilities.hpp"
+
+namespace axom
+{
+namespace klee
+{
+void ShapeSet::setShapes(std::vector<Shape> shapes)
+{
+  m_shapes = std::move(shapes);
+}
+
+void ShapeSet::setPath(const std::string &path) { m_path = path; }
+
+std::string ShapeSet::resolvePath(const std::string &filePath) const
+{
+  if(m_path.empty())
+  {
+    throw std::logic_error("The ShapeSet's path has not been set");
```

Comment:
`KleeError` here as well?

Reply:
We throw a `KleeError` in `IOError.cpp` (line 159) when validating input. `KleeError` is used to report input errors by the user. This instance is a logic error by the programmer.